rules/new-for-builtins.js

```diff
@@ -33,11 +33,36 @@ const disallowNew = new Set([
 	'Symbol'
 ]);
 
+function isVariableInScope(context, name) {
```

Comment:
So, if I got this right, this function is supposed to check whether the variable is defined by the user (returns `true` in that case), or the variable is implicitly defined in the global scope. Is this right?

If so, here is an alternative implementation idea, without having to collect all the variables into an array:

<p>
<details>
 <summary>Click to see, or don't click and try to figure it out :thinking: </summary>

Walk up the scopes (via `scope.upper`) while checking each scope's [`.set`](https://eslint.org/docs/developer-guide/scope-manager-interface#set) for the variable name. If you find it and the scope is not the global scope, the variable was defined by the user, otherwise it came from the global scope.

</details>
</p>


Also, if I did get the intention right, a better name for the function would be `isUserDefinedVariable`.

Reply:
And I think it should be moved to the `utils` directory.

Reply:
This is already implemented in https://github.com/sindresorhus/eslint-plugin-unicorn/blob/master/rules/utils/resolve-variable-name.js